<a href="https://colab.research.google.com/github/harshit00052/DecodeLabs-Internship/blob/main/internship.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd


In [ ]:
df = pd.read_csv('/content/Dataset for Data Analytics - Sheet1.csv')
df.sample(5)
# df.describe()
# df.shape

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
1046,ORD201046,2024-10-07,C85421,Tablet,3,547.15,466 Main St,Online,Shipped,TRK16524006,4,WINTER15,Instagram,1641.45
1018,ORD201018,2023-07-08,C77499,Laptop,3,182.48,209 Main St,Cash,Delivered,TRK20703086,3,NaN,Instagram,547.44
791,ORD200791,2024-01-13,C49710,Chair,4,409.28,956 Main St,Cash,Pending,TRK92591237,8,NaN,Email,1637.12
136,ORD200136,2024-04-27,C73370,Laptop,1,44.35,595 Main St,Gift Card,Cancelled,TRK67820278,1,SAVE10,Instagram,44.35
731,ORD200731,2024-06-07,C71310,Printer,5,584.46,308 Main St,Credit Card,Returned,TRK48292586,8,NaN,Email,2922.30


1) analysed the raw dataset using df.info()

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   OrderID          1200 non-null   object 
 1   Date             1200 non-null   object 
 2   CustomerID       1200 non-null   object 
 3   Product          1200 non-null   object 
 4   Quantity         1200 non-null   int64  
 5   UnitPrice        1200 non-null   float64
 6   ShippingAddress  1200 non-null   object 
 7   PaymentMethod    1200 non-null   object 
 8   OrderStatus      1200 non-null   object 
 9   TrackingNumber   1200 non-null   object 
 10  ItemsInCart      1200 non-null   int64  
 11  CouponCode       891 non-null    object 
 12  ReferralSource   1200 non-null   object 
 13  TotalPrice       1200 non-null   float64
dtypes: float64(2), int64(2), object(10)
memory usage: 131.4+ KB


2) looking for null values in columns using isnull() -> (result) found CouponCode column have 309 null columns

In [ ]:
df.isnull().sum()

,0
OrderID,0
Date,0
CustomerID,0
Product,0
Quantity,0
UnitPrice,0
ShippingAddress,0
PaymentMethod,0
OrderStatus,0
TrackingNumber,0


3)  filled the null or blank rows in CouponCode column with no coupon

In [ ]:
df['CouponCode'] = df['CouponCode'].fillna('No Coupon')

4) looking for duplicate rows (specially in orderID Col) in dataset using duplicated() and nuniue() -> (result) found 0 duplicate rows

In [ ]:
df[df.duplicated(keep=False)]
# df['OrderID'].nunique()
df['OrderID'].duplicated().sum()
# df['TrackingNumber'].duplicated().sum()

np.int64(0)

5) changed the datatype of date from object to datetime64ns

In [ ]:
df['Date'] = pd.to_datetime(df['Date'])

6) changed the datatype of Quantiy and ItemInCart from int64 to int32 as don't need it reduced the size of dataset from 131.4 kb to 122 kb

In [ ]:
df['Quantity'] = df['Quantity'].astype(np.int32)
df['ItemsInCart'] = df['ItemsInCart'].astype(np.int32)


7) reanalysed the data

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int32         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int32         
 11  CouponCode       1200 non-null   object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int32(

In [ ]:
df.isnull().sum()

,0
OrderID,0
Date,0
CustomerID,0
Product,0
Quantity,0
UnitPrice,0
ShippingAddress,0
PaymentMethod,0
OrderStatus,0
TrackingNumber,0
